### PR TITLE
Feature/accessible drafts

### LINKED ISSUE #409: Make it easy to find drafts and submit them
#### Description

As a user i am not able to see my drafts and follow through with the submissions.

We can move the `Choose from drafts`
image

on the right with the submitted objects under different sections
image
We can also add to the button group there `Submit` so that we have `Submit, Continue, Delete`

**Note: other ideas are welcomed**

#### DoD (Definition of Done)

Move drafts to a location where users can easily find them

#### Testing

Unit and Integration testing

### PR DESCRIPTION
**Pull request by saulipurhonen**
### Description

Moved draft objects from `Object index` to right sidebar. Draft objects are listed below `Form` & `XML` submissions.

### Related issues

Closes #409 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- Use MUI Grid layout on add object -step
- Refactored `submittedObjectsList` component to work with drafts
- Removed `Choose from drafts` option from object index
- Created custom hook for object submit functionality
- Refactored integration and E2E tests

### Testing

- [x] Integration Tests
- [x] E2E

### CODE REVIEW

File: src/components/NewDraftWizard/WizardComponents/WizardSavedObjectsList.js [hunk 1]
```diff
@@ -70,35 +69,41 @@ const WizardSavedObjectsList = ({ submissions }: WizardSavedObjectsListProps): R
...
 
   return (
     <div className={classes.objectList}>
-      {groupedSubmissions &&
-        groupedSubmissions.map(group => (
+      {listItems &&
+        listItems.map(group => (
           <Box pt={0} key={group.submissionType}>
             <CardHeader
               title={`Submitted ${formatDisplayObjectType(objectType)} ${displaySubmissionType(group)}`}
```
Comment (blankdots) [inline src/components/NewDraftWizard/WizardComponents/WizardSavedObjectsList.js:101]: not sure if this is where it is, but looks like it, maybe for drafts we want other label than `Submitted`

image

maybe `<object> Drafts` e.g. `Sample Drafts` and the colour of the background of title could be different to emphasize that this is different than the submitted objects

**Review by blankdots — CHANGES_REQUESTED** [review]
I did notice at some point some weird behaviour when going back and forth between editing a draft and submitting it but was not able to reproduce 2nd time - mentioning in case someone else notices.

**Comment by blankdots** [comment]
A suggestion of improvement could be to add some sort of `spinner` when on the right when that list updates e.g. when a draft moves from draft to submitted to change is too sudden and it disappears without warning

File: src/components/NewDraftWizard/WizardComponents/WizardSavedObjectsList.js (hunk 1 above)
Comment (blankdots) [reply src/components/NewDraftWizard/WizardComponents/WizardSavedObjectsList.js:101]: or we can have something like
- `Sample Submitted Forms`
- `Sample Submitted XML files`
- `Sample Drafts`

File: src/components/NewDraftWizard/WizardComponents/WizardSavedObjectActions.js [hunk 2]
```diff
@@ -27,6 +28,7 @@ const useStyles = makeStyles(theme => ({
 }))
 
 type WizardSavedObjectActionsProps = {
+  folderId: string,
```
Comment (hannyle) [inline src/components/NewDraftWizard/WizardComponents/WizardSavedObjectActions.js:31]: It seems that this `folderId` is not used anywhere in the file?

**Comment by saulipurhonen** [comment]
> the only change i request is with the labels to distinguish between drafts and submitted objects better.

Background color of drafts title and title text changed:
image

> I did notice at some point some weird behaviour when going back and forth between editing a draft and submitting it but was not able to reproduce 2nd time - mentioning in case someone else notices.

Let's keep eyes open on this and make a bug issue if found after merge.

File: src/components/NewDraftWizard/WizardComponents/WizardSavedObjectActions.js (hunk 2 above)
Comment (saulipurhonen) [reply src/components/NewDraftWizard/WizardComponents/WizardSavedObjectActions.js]: Good call. This prop was used in early version for this feature. Removed on latest commit.

**Comment by saulipurhonen** [comment]
> A suggestion of improvement could be to add some sort of `spinner` when on the right when that list updates e.g. when a draft moves from draft to submitted the change is too sudden and it disappears without warning
> 
> https://material-ui.com/components/progress/#circular

I added a MUI Circular Progress indicator when drafts are saved or objects submitted.

**Review by blankdots — APPROVED** [review]

**Review by hannyle — APPROVED** [review]
LGTM.

**Pull request merged**